jvm-libs/linea/clients/linea-l1-contract-client/src/main/kotlin/build/linea/contract/l1/Web3JLineaRollupSmartContractClientReadOnly.kt

```diff
@@ -73,7 +72,9 @@ open class Web3JLineaRollupSmartContractClientReadOnly(
     } else {
       fetchSmartContractVersion()
         .thenPeek { contractLatestVersion ->
-          if (contractLatestVersion != smartContractVersionCache.get()) {
+          if (smartContractVersionCache.get() != null &&
+            contractLatestVersion != smartContractVersionCache.get()
+          ) {
```

Comment:
If I read the code correctly, this change:

1 - it's not complete. 
2 - it introduces a Bug: `smartContractVersionCache` starts null, and it's only changed on line 84 if this if is true. So it will remain null forever.

Reply:
No, this condition just controls whether the "upgraded" log should be printed or not, the idea is not to print it if it was changed from null.

Reply:
That is correct. I misread the indentation. My appologies